### PR TITLE
feat(vpn): add new data_source gateway_flavors

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/vpn' TESTARGS='-run TestAccDatasourceVpnGatewayFlavors_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccDatasourceVpnGatewayFlavors_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceVpnGatewayFlavors_basic
=== PAUSE TestAccDatasourceVpnGatewayFlavors_basic
=== CONT  TestAccDatasourceVpnGatewayFlavors_basic
--- PASS: TestAccDatasourceVpnGatewayFlavors_basic (32.92s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       32.961s
```

### CODE REVIEW

File: docs/data-sources/vpn_gateway_flavors.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Use this data source to get the list of VPN gateway flavors.
+
+## Example Usage
+
+```hcl
+variable "name" {}
+
+data "huaweicloud_vpn_gateway_flavors" "test"{
+  availability_zone = "cn-north-4"
```
Comment (ShiChangkuo) [inline docs/data-sources/vpn_gateway_flavors.md:15]:
```suggestion
  availability_zone = "cn-north-4a"
```

File: docs/data-sources/vpn_gateway_flavors.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+* `name` - (Optional, String) Specifies the flavor name.
+
+* `attachment_type` - (Optional, String) Specifies the attachment type. The value can be: **vpc** and **er**.
+  Defaults to **vpc**.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
```
Comment (ShiChangkuo) [inline docs/data-sources/vpn_gateway_flavors.md:38]:
```suggestion
* `id` - The data source ID.

```

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_gateway_flavors.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+func filterFlavors(all interface{}, d *schema.ResourceData) []interface{} {
+	availabilityZone := d.Get("availability_zone").(string)
+	attachmentType := d.Get("attachment_type").(string)
+	allMap := all.(map[string]interface{})
+	rst := make([]interface{}, 0, len(allMap))
+
+	for k, v := range allMap {
+		if flavorName, ok := d.GetOk("name"); ok && flavorName.(string) != k {
```
Comment (ShiChangkuo) [inline huaweicloud/services/vpn/data_source_huaweicloud_vpn_gateway_flavors.go:119]: it's better to get flavorName outside of the for loop

File: huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_gateway_flavors_test.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+				),
+			},
+		},
+	})
+}
+
+func testAccDatasourceVpnGatewayFlavors_basic() string {
+	return `
+data "huaweicloud_vpn_gateway_flavors" "test" {
+  availability_zone = "cn-north-4a"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_gateway_flavors_test.go:37]: suggest to use `data.huaweicloud_availability_zones`

**Comment by Jason-Zhang9309** [comment]
redesign to huaweicloud_vpn_gateway_availability_zones

**Pull request closed without merging**